src/processTargets/modifiers/ModifyIfUntypedStage.ts

```diff
@@ -1,38 +1,77 @@
 import { Target } from "../../typings/target.types";
-import { ModifyIfUntypedModifier } from "../../typings/targetDescriptor.types";
+import {
+  Modifier,
+  ModifyIfUntypedModifier,
+} from "../../typings/targetDescriptor.types";
 import { ProcessedTargetsContext } from "../../typings/Types";
 import getModifierStage from "../getModifierStage";
 import { ModifierStage } from "../PipelineStages.types";
 
-/**
- * Runs {@link ModifyIfUntypedModifier.modifier} if the target has no explicit
- * scope type, ie if {@link Target.hasExplicitScopeType} is `false`.
- */
-export default class ModifyIfUntypedStage implements ModifierStage {
+abstract class ModifyIfBaseStage implements ModifierStage {
   private nestedStage_?: ModifierStage;
 
-  constructor(private modifier: ModifyIfUntypedModifier) {}
+  constructor(private modifier: Modifier, private suppressErrors?: boolean) {}
 
   run(context: ProcessedTargetsContext, target: Target): Target[] {
-    // If true this target has an explicit scope type and should not be modified.
-    if (target.hasExplicitScopeType) {
-      return [target];
+    if (this.modifyIf(target)) {
+      // Modify this target
+      try {
+        return this.nestedStage
+          .run(context, target)
+          .map((newTarget) => newTarget.withThatTarget(target));
+      } catch (ex) {
+        // suppressErrors === true => Allow this target to be returned unmodified
+        if (!this.suppressErrors) {
+          throw ex;
+        }
+      }
     }
 
-    /**
-     * This target is lacking an explicit scope type and should use inference/upgrade when needed.
-     * See {@link Target.hasExplicitScopeType} for more info
-     */
-    return this.nestedStage
-      .run(context, target)
-      .map((newTarget) => newTarget.withThatTarget(target));
+    // Don't modify this target
+    return [target];
   }
 
   private get nestedStage() {
     if (this.nestedStage_ == null) {
-      this.nestedStage_ = getModifierStage(this.modifier.modifier);
+      this.nestedStage_ = getModifierStage(this.modifier);
     }
 
     return this.nestedStage_;
   }
+
+  protected abstract modifyIf(target: Target): boolean;
+}
+
+/**
+ * Runs {@link ModifyIfUntypedModifier.modifier} if the target has no explicit
+ * scope type, ie if {@link Target.hasExplicitScopeType} is `false`.
+ */
+export class ModifyIfUntypedStage extends ModifyIfBaseStage {
+  constructor(modifier: ModifyIfUntypedModifier) {
+    super(modifier.modifier);
+  }
+
+  protected modifyIf(target: Target): boolean {
+    return !target.hasExplicitScopeType;
+  }
+}
+
+/**
+ * Tries to convert target into token if:
+ * {@link Target.hasExplicitScopeType} is `false` and
+ * {@link Target.hasExplicitRange} is `false` and
+ * {@link Target.contentRange.isEmpty} is `true`.
+ */
+export class ModifyUnTypedEmptyToTokenStage extends ModifyIfBaseStage {
```

Comment:
Capitalisation here is inconsistent with `modifyIfUntyped`: `UnTyped` should be `Untyped`.  Actually this name is a bit wonky overall.  Maybe `ContainingTokenIfUntypedEmptyStage`?